[RFC] Adding functionality to handle PUK code

hasnainvirk opened the pull request:
### Description

Its annoying to unlock SIM cards if they get stuck for PUK code
authentication. Too many wrong attempts to set the PIN code would result
in PUK lock situation. Previously, we had state handling for PUK
requirement but there was no code to actually provide it on runtime.


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [X] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

Commit:
Adding functionality to handle PUK code

Its annoying to unlock SIM cards if they get stuck for PUK code
authentication. Too many wrong attempts to set the PIN code would result
in PUK lock situation. Previously, we had state handling for PUK
requirement but there was no code to actually provide it on runtime.

jarvte commented on features/cellular/framework/AT/AT_CellularDevice.cpp:
```diff
@@ -163,6 +163,24 @@ nsapi_error_t AT_CellularDevice::get_sim_state(SimState &state)
+    }
+
+    _at->lock();
+    _at->set_at_timeout(5000);
```
if timeout is changed, it must be set back with _at->restore_at_timeout();

hasnainvirk replied:
@jarvte I don't think so. The prototype of the function is  `set_at_timeout(uint32_t timeout, bool default_timeout = false) `. So by default it wouldn't store the timeout as default.

hasnainvirk replied:
Ah, it doesn't restore after use.  The doc needs to be added perhaps.

hasnainvirk replied:
cmd_stop_read_resp() method should restore the timeout logically if default_timeout was false in set_at_timeout(...)

Commit:
Restoring AT timeout to default explicitly

By default set_at_timeout(...) should not set timeout to default
timeout. It doesn't but after the command is complete nobody restores
the default timeout which should have been the case as per doc. So we
have to explicitly restore it.

Commit:
Astyle issues rectified

Minor astyle issues addressed.
